__tests__/app.test.js

```diff
@@ -86,6 +86,63 @@ describe('Test GET api/articles  endpoint', () => {
         })
       })
   })
+  test('Test queries sort_by and order in ascending order', () => {
+    return request(app)
+      .get('/api/articles?sort_by=topic&order=asc')
+      .expect(200)
+      .then(({body}) => {
+        expect(body.length === 13).toBe(true)
+        expect(body).toBeSortedBy('topic', {
+          descending: false
+        })
+      })
+  })
+  test('Test queries sort_by and order by comment_count', () => {
+    return request(app)
+      .get('/api/articles?sort_by=comment_count&order=asc')
+      .expect(200)
+      .then(({body}) => {
+        expect(body.length === 13).toBe(true)
+        expect(body).toBeSortedBy('comment_count', {
+          descending: false
+        })
+      })
+  })
+  test('Test queries [topic, sort_by and order]', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch&sort_by=comment_count&order=asc')
+      .expect(200)
+      .then(({body}) => {
+        expect(body.length === 12).toBe(true)
+        expect(body).toBeSortedBy('comment_count', {
+          descending: false
+        })
+      })
+  })
+  test('Test queries [topic, sort_by and order] but no topic is banana so returns empty array', () => {
+    return request(app)
+      .get('/api/articles?topic=banana&sort_by=comment_count&order=asc')
+      .expect(200)
+      .then(({body}) => {
+        expect(body).toEqual([])
+      })
+  })
+  test('Test queries [topic, sort_by and order] but sort_by is incorrect', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch&sort_by=nothing&order=asc')
+      .expect(404)
+      .then(({body}) => {
+        expect(body).toEqual({msg:"Incorrect sort_by"})
+      })
+  })
+  test('Test queries [topic, sort_by and order] but order is incorrect', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch&sort_by=topic&order=banana')
+      .expect(404)
+      .then(({body}) => {
+        expect(body).toEqual({msg:"Incorrect order"})
+      })
+  })
```

Comment:
Great effort on the testing here. I know there's a lot to cover! Some tests I would add here:
- a 200 for when a valid topic is passed but it has no articles yet (I think the topic in our data for this would be 'paper')
- you seem to have passed asc for all your order by, so don't be afraid to pass desc for a few too 😄  on the topic of asc / desc, I would add a test that checks that the default order works as well as the default sort_by. As mentioned above, it's nicer to have some individual query tests as these allow us to make sure the defaults are working. Remember, we expect descending order, sorted by created_by for the default